node/pkg/wfetcher/common/utils.go

```diff
@@ -0,0 +1,88 @@
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"math"
+	"strconv"
+	"strings"
+
+	"bisonai.com/orakl/node/pkg/db"
+	"github.com/rs/zerolog/log"
+)
+
+/*
+Generates two types of maps with different keys for the same feed ID.
+The combined map has keys like "BTCUSD", and the separated map has keys like "BTC-USD".
+The reason for this is that different exchanges use different naming conventions.
+*/
+func GetWssFeedMap(feeds []Feed) map[string]FeedMaps {
+	feedMaps := make(map[string]FeedMaps)
+	for _, feed := range feeds {
+		if !strings.Contains(feed.Name, "wss") {
+			continue
+		}
+
+		raw := strings.Split(feed.Name, "-")
+		if len(raw) != 4 {
+			log.Warn().Str("name", feed.Name).Msg("invalid name")
+			continue
+		}
+
+		provider := strings.ToLower(raw[0])
+		base := strings.ToUpper(raw[2])
+		quote := strings.ToUpper(raw[3])
+		combinedName := base + quote
+		separatedName := base + "-" + quote
+
+		if _, exists := feedMaps[provider]; !exists {
+			feedMaps[provider] = FeedMaps{
+				Combined:  make(map[string]int32),
+				Separated: make(map[string]int32),
+			}
+		}
+		feedMaps[provider].Combined[combinedName] = feed.ID
+		feedMaps[provider].Separated[separatedName] = feed.ID
+	}
+	return feedMaps
+}
+
+func StoreFeeds(ctx context.Context, feedData []FeedData) error {
+	latestData := make(map[string]any)
+	for _, data := range feedData {
+		key := "latestFeedData:" + strconv.Itoa(int(data.FeedId))
```

Comment:
`"latestFeedData:X" is used at multiple places (https://github.com/search?q=repo%3ABisonai%2Forakl%20latestFeedData%3A&type=code). it does not seem to be parsed directly anywhere, but if we decide to use it for some purpose, it would be nice to have a single place where this key is generated.

Reply:
yes, totally agreed, I'll proceed with that refactoring soon and related [issue](https://github.com/Bisonai/orakl/issues/1536)  has been made